src/server.tsx

```diff
@@ -161,12 +166,18 @@ export async function serveFragment(
       <Component {...childProps} />
     </div>
   );
+
   const removeReactRootStream = new RemoveReactRoot();
   stream.pipe(
     removeReactRootStream,
     { end: false }
   );
-  removeReactRootStream.pipe(
+
+  let lastStream: NodeJS.ReadableStream = removeReactRootStream;
```

Comment:
```suggestion
  const lastStream: NodeJS.ReadableStream =  options.pipeStream ? removeReactRootStream : options.pipeStream(removeReactRootStream);
```

Reply:
Thanks that is cleaner indeed. I've added it in commit 047a3db:
```
const lastStream: NodeJS.ReadableStream =  options.pipeStream ? options.pipeStream(removeReactRootStream) : removeReactRootStream;
```